### ISSUE TITLE
Automatic casting of *64 values

### ISSUE THREAD

**Issue by bdice** [glotzerlab/freud#98]
**[Original report](https://bitbucket.org/glotzer/freud/issue/98) by Eric Harper (Bitbucket: [harperic](https://bitbucket.org/harperic), GitHub: [harperic](https://github.com/harperic)).**

----------------------------------------

We should automatically cast np.float64, etc. values to np.float32, etc. values.

**Comment by bdice** [comment]
**Original comment by Eric Harper (Bitbucket: [harperic](https://bitbucket.org/harperic), GitHub: [harperic](https://github.com/harperic)).**

----------------------------------------

Possible Soltions:

1. Leave behavior as is, and make the error message more useful:

```
#!python

raise RuntimeError("positions must be of type np.float32. Please use np.asarray(pos, dtype=np.float32) or pos.astype(np.float32) to cast" 
```

2. Template all functions so that both 32 and 64 are able to be used. Problem with this is that the memory usage and speed benefits would disappear for most users as numpy defaults float64, etc.

3. Template all functions, but use cython to wrap as follows:


```
#!python

def accumulate(box, pos, useDouble=False):
    if (pos.dtype != np.float32):
        if (pos.dtype != np.float64):
            raise ValueError()
        elif not useDouble:
            raise RuntimeWarning("Casting pos to float32. Please use useDouble=True to use double precision calculation")
            pos = pos.astype(dtype=np.float32)
```

**Comment by bdice** [comment]
**Original comment by Eric Harper (Bitbucket: [harperic](https://bitbucket.org/harperic), GitHub: [harperic](https://github.com/harperic)).**

----------------------------------------

Fixed with PR #94
